tests/writer_test.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""Tests for the artifact definitions readers."""
+
+import io
+import os
+import sys
+import unittest
+import yaml
+import json
+
+from artifacts import errors
+from artifacts import reader
+from artifacts import writer
+
+
+class ArtifactAsDictTest(unittest.TestCase):
+  """Class to test the artifacts AsDict conversion"""
+
+  def testAsDict(self):
+    """Tests the ArtifactDefinition AsDict method."""
+    artifact_reader = reader.YamlArtifactsReader()
+    test_file = os.path.join('test_data', 'definitions.yaml')
+
+    with open(test_file, 'r') as file_object:
+      for artifact_definition in yaml.safe_load_all(file_object):
+        artifact_object = artifact_reader._ReadArtifactDefinition(
+            artifact_definition)
+        self.assertEqual(artifact_definition, artifact_object.AsDict())
+
+  def testDefinitionsAsDict(self):
+    """Tests that all defined artifacts can convert to dictionary representation."""
+    artifact_reader = reader.YamlArtifactsReader()
+
+    artifact_definitions = list(artifact_reader.ReadDirectory('definitions'))
+
+    last_artifact_definition = None
+    for artifact in artifact_definitions:
+      try:
+        artifact_definition = artifact.AsDict()
+      except errors.FormatError:
+        error_location = u'At start'
+        if last_artifact_definition:
+          error_location = u'After: {0}'.format(last_artifact_definition.name)
+        self.fail(u'{0} failed to convert to dict'.format(error_location))
```

Comment:
Related, should this just be except: instead of except errors.FormatError since the test is that it should be able to run AsDict on everything in definitions and not fail for any reason.